lib/segment/src/segment.rs

```diff
@@ -1384,6 +1411,22 @@ impl SegmentEntry for Segment {
         }
     }
 
+    fn read_random_filtered(&self, limit: usize, filter: Option<&Filter>) -> Vec<PointIdType> {
+        match filter {
+            None => self.read_by_random_id(limit),
+            Some(condition) => {
+                if self.should_pre_filter(condition, Some(limit)) {
+                    let mut points = self.filtered_read_by_index(None, Some(limit), condition);
+                    let mut rng = thread_rng();
+                    points.shuffle(&mut rng);
```

Comment:
If I read this correctly, we always return the same set of points in case of filter, but just in a different order.

We could also increase randomly the `limit` to try to surface different points at each call.

Reply:
Yeah, that I missed to think of that. Fixed in https://github.com/qdrant/qdrant/pull/4619/commits/f79d705dd0deadea16ce67b461f31d706f2d7c6b